src/components/GoogleLoginButton.js

```diff
@@ -0,0 +1,27 @@
+import React from 'react';
+
+export default function GoogleLoginButton({ setUserInfo }) {
+  const baseUrl = process.env.NEXT_PUBLIC_API_URL;
+
+  async function signInGoogle() {
+    const { user } = await firebase.auth().signInWithPopup(new firebase.auth.GoogleAuthProvider());
+    const userInfo = {
+      email: user.email,
+      name: user.displayName,
+      photoUrl: user.photoURL,
+    };
+
+    const { data } = await fetch(`${baseUrl}/login`, {
+      method: 'POST',
+      body: JSON.stringify(userInfo),
+    });
```

Comment:
이렇게 해도 될것 같네요. 불필요한 `userInfo` 변수 선언을 줄일 수 잇어 보입니다.

```suggestion
    const { email, displayName: name, photoURL: photoUrl } = await firebase.auth().signInWithPopup(new firebase.auth.GoogleAuthProvider());

    const { data } = await fetch(`${baseUrl}/login`, {
      method: 'POST',
      body: JSON.stringify({ email, name photoUrl }),
    });
```